rudder-web/src/main/scala/com/normation/rudder/web/rest/node/NodeAPIService8.scala

```diff
@@ -138,54 +128,32 @@ class NodeApiService8 (
     }
   }
 
-  private[this] class RunHandler (out : Option[OutputStream]) extends NuAbstractProcessHandler {
-
-    // set output (stdout or stderr) buffer to our stream
-    // Synchronized so outputs are not mixed
-    def onOut(buf: ByteBuffer, isClosed : Boolean) = {
-      out match {
-        case Some(out) =>
-          val bytes : Array[Byte] = new Array(buf.remaining())
-          buf.get(bytes)
-          out.synchronized {
-            out.write(bytes)
-            out.flush()
-          }
-        case None => // Nothing to do for now
-      }
-    }
-
-    // On exit will be called at the end of the process, or if any error/exception occurs, so we can close our outputstream here
-    override def onExit( statusCode : Int) = {
-      out.map(_.close())
-    }
+  def remoteRunRequest(nodeId: NodeId, classes : List[String], keepOutput : Boolean, asynchronous : Boolean) = {
+    val url = s"https://localhost/rudder/relay-api/remote-run/nodes/${nodeId.value}"
+    val params =
+      ("classes"      , classes.mkString(",") ) ::
+      ( "keep_output" , keepOutput.toString   ) ::
+      ( "asynchronous", asynchronous.toString ) ::
+      Nil
 
-    override def onStdout(buf: ByteBuffer, isClosed : Boolean) = onOut(buf,isClosed)
-
-    override def onStderr(buf: ByteBuffer, isClosed : Boolean) = onOut(buf,isClosed)
+    Http(url).params( params )
   }
 
   def runNode(nodeId: NodeId, classes : List[String]) : Box[OutputStream => Unit] = {
 
+    val request = remoteRunRequest(nodeId,classes,true,false)
     import net.liftweb.util.Helpers.tryo
     for {
-      node <- nodeInfoService.getNodeInfo(nodeId).flatMap {node => Box(node) ?~! s"Could not find node '${nodeId.value}' informations" }
-      in = new PipedInputStream(pipeSize)
-      out = new PipedOutputStream(in)
-      _ <- try {
-             import scala.collection.JavaConversions._
-             val pb = new NuProcessBuilder("/usr/bin/rudder" :: runAction(node) ::: classOption(classes))
-             val handler = new RunHandler(Some(out))
-             pb.setProcessListener(handler)
-             Full(pb.start())
-           } catch {
-             case e : Throwable =>
-               out.close()
-               Failure(s"An error occured when applying policy on Node '${node.id.value}', cause is: ${e.getMessage}")
-           }
-    } yield {
-      runResponse(in)
-    }
+       httpResponse <- tryo { request.execute { runResponse } }
+       response <- if (httpResponse.isSuccess ) {
+         Full(httpResponse.body)
```

Comment:
request.execute allow to get data from the response as a stream ( inputStrem -> T ) we transform it using "runResponse" to an "OutputStream => Unit" which will read from the inputstream to put into the api output stream response and stream it


so yeah, body is not the "response data" but "how you want to get the response data" ... and it's quite misleading